mmv1/third_party/validator/project.go

```diff
@@ -125,24 +126,17 @@ func GetProjectBillingInfoCaiObject(d TerraformResourceData, config *Config) ([]
 	}
 }
 
-func GetProjectBillingInfoApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+func GetProjectBillingInfoApiObject(d TerraformResourceData, project string) (map[string]interface{}, error) {
 	if _, ok := d.GetOk("billing_account"); !ok {
 		// TODO: If the project already exists, we could ask the API about it's
 		// billing info here.
 		return nil, ErrNoConversion
 	}
 
-	var name string
-	if number, ok := d.GetOk("number"); ok {
-		name = fmt.Sprintf("projects/%s/billingInfo", number)
-	} else {
-		name = fmt.Sprintf("projects/%s/billingInfo", d.Get("project_id"))
-	}
-
 	ba := &cloudbilling.ProjectBillingInfo{
 		BillingAccountName: fmt.Sprintf("billingAccounts/%s", d.Get("billing_account")),
-		Name:               name,
-		ProjectId:          d.Get("project_id").(string),
+		Name:               fmt.Sprintf("projects/%s/billingInfo", project),
+		ProjectId:          project,
```

Comment:
It will always be what assetName has. `//cloudbilling.googleapis.com/projects/{{X}}/billingInfo`, it will always be the X. 

However, around line 56, there is 

```
func GetProjectApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
	pid := d.Get("project_id").(string)
	project := &cloudresourcemanager.Project{
		ProjectId: pid,
		Name:      d.Get("name").(string),
	}
```
Would that also need to use the same as the asset name?

Reply:
In both cases, they should match what is returned by CAI - could you double-check what that is? If both are supposed to be project id, then they should always be project id (rather than matching what is in the asset name, which might be the project number instead.)

Reply:
The cloudresourcemanager.Project is defined as such: 
```
	// ProjectId: The unique, user-assigned ID of the Project. It must be 6
	// to 30 lowercase letters, digits, or hyphens. It must start with a
	// letter. Trailing hyphens are prohibited. Example: `tokyo-rain-123`
	// Read-only after creation.
	ProjectId string `json:"projectId,omitempty"`

	// ProjectNumber: The number uniquely identifying the project. Example:
	// `415104041262` Read-only.
	ProjectNumber int64 `json:"projectNumber,omitempty,string"`
```
If d.Get("project_id") == "", this just leaves it as empty?

Reply:
Just updated the code to restore to d.Get("project_id") here, and updated the test cases.